### PR TITLE
doc: Plugins Documentation

### PR DESCRIPTION
**Pull request by IronCore864**
# Summary

## Key Points

- [ ] This is a breaking change.
- [x] Documentation (new or existing) is updated.

## Description

Plugins documentation updated:

- how to create a plugin
- doc for GitHub Actions plugin
- doc for ArgoCD plugin
- doc for ArgoCD App plugin

### Related Issues

https://github.com/merico-dev/stream/issues/6

### CODE REVIEW

**Comment by IronCore864** [comment]
@lfbdev @daniel-hutao, please help to review this change.

Please pay close attention to grammar, syntax, the example usage, and the comments in it.

You can commit directly to this branch when you find a bug.

Merge and delete the branch after reviewing is done. Thanks.

**Comment by IronCore864** [comment]
Feel free to enrich the content, too.

File: docs/argocd_app_plugin.md [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+      name: argo/argo-cd
+      release_name: argocd
+      namespace: argocd
+      create_namespace: False
+# name of the plugin
+- name: argocdapp
+  # version of the plugin
+  version: 0.0.1
+  # options for the plugin
+  # Checkout the version from the GitHub releases.
```
Comment (daniel-hutao) [inline docs/argocd_app_plugin.md:30]: We could change "Checkout" to lowercase "checkout" to keep it consistent with the previous comment.

**Review by daniel-hutao — CHANGES_REQUESTED** [review]
done

File: docs/argocd_app_plugin.md (hunk 1 above)
Comment (daniel-hutao) [reply docs/argocd_app_plugin.md]: #51

**Pull request merged**